core/internal/runhistoryreader/parquet/iterator/record_iterator.go

```diff
@@ -0,0 +1,187 @@
+package iterator
+
+import (
+	"fmt"
+
+	"github.com/apache/arrow-go/v18/arrow"
+)
+
+type keyIteratorPair struct {
+	Key      string
+	Iterator columnIterator
+}
+
+// recordIterator iterates over columnIterators
+// to read columns in the given arrow.RecordBatch in chunks.
+type recordIterator struct {
+	record           arrow.RecordBatch
+	columns          []keyIteratorPair
```

Comment:
If a column is not one of the `resultCols` and not the `indexCol`, do we really need it? Why have this field?